src/components/Forms/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  CssBaseline,
+  Typography,
+  Container,
+  Button,
+  TextField,
+} from '@material-ui/core';
+import cookies from 'react-cookies';
+import ForumRoundedIcon from '@material-ui/icons/ForumRounded';
+import config from '../../../config';
+import { serviceRequest } from '../../../services/serviceRequest';
+import useStyles from './style';
+
+
+const path = '/api/v1/signup';
+const domain = config.api_domain;
+
+export const SignUpForm = () => {
+  const classes = useStyles();
+  const [formData, setFormData] = useState({
+    email: '',
+    username: '',
+    password: '',
+  });
+  const [isSignedUp, setIsSignedUp] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+
+  function handleIsSignedUp() {
+    if (isSignedUp) {
+      return <Redirect to="/tokenVerify" />;
+    }
+    return null;
+  }
+
+  function handleIsError() {
+    if (isError) {
+      return <p>{errorMsg}</p>;
+    }
+    return null;
+  }
+
+  function handleChange(event) {
+    const updatedForm = { ...formData };
+    updatedForm[event.target.name] = event.target.value;
+    setFormData(updatedForm);
+  }
+
+  function preprarePayload(method, data) {
+    const url = `https://${domain}${path}`;
+    return {
+      method,
+      url,
+      data,
+    };
+  }
+
+  async function handleSubmit(e) {
+    try {
+      e.preventDefault();
+      const reqInfo = preprarePayload('post', formData);
+      const response = await serviceRequest(reqInfo);
+      if (response.status && response.status === 'success') {
+        const { data } = response;
+        const jwtToken = `bearer ${data.token}`;
+        const expirationTime = 60 * 60 * 24; //  24 hours
+        cookies.save('jwtToken', jwtToken, { path: '/', maxAge: expirationTime });
+        setIsSignedUp(true);
+      } else if (response.status && response.status === 'error') {
+        setErrorMsg('Invalid email, username or password');
```

Comment:
maybe you can extract the error message from the response data, most of the time, fail to signup is because of duplicate email or username and should let user know about it